src/screencast/xdpw_gbm.c

```diff
@@ -0,0 +1,56 @@
+#include <gbm.h>
+#include <xf86drm.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "xdpw.h"
+#include "logger.h"
+
+static char *gbm_find_render_node(size_t maxlen) {
+	drmDevice *devices[64];
+	char *render_node = NULL;
+
+	int n = drmGetDevices2(0, devices, sizeof(devices) / sizeof(devices[0]));
+	for (int i = 0; i < n; ++i) {
+		drmDevice *dev = devices[i];
+		if (!(dev->available_nodes & (1 << DRM_NODE_RENDER)))
+			continue;
+
+		render_node = strndup(dev->nodes[DRM_NODE_RENDER], maxlen);
+		break;
+	}
+
+	drmFreeDevices(devices, n);
+	return render_node;
+}
+
+struct gbm_device *create_gbm_device(){
```

Comment:
Style: missing space between `)` and `{`

Should use `(void)` to make sure the caller doesn't pass unexpected args.